Microsoft.Toolkit.Uwp.UI.Controls.Primitives/ConstrainedPresenter/ConstrainedPresenter.cs

```diff
@@ -0,0 +1,119 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.Foundation;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// The <see cref="ConstrainedPresenter"/> is a <see cref="ContentPresenter"/> control which can restrict the
+    /// available size for its content based on a scale factor and/or a specific <see cref="AspectRatio"/>.
+    /// </summary>
+    public class ConstrainedPresenter : ContentPresenter // TODO: Think it should be a Border? But it's sealed; hopefully can change in WinUI 3.
+    {
+        /// <summary>
+        /// Gets or sets aspect Ratio to use for the contents of the Panel (after scaling).
+        /// </summary>
+        public AspectRatio AspectRatio
+        {
+            get { return (AspectRatio)GetValue(AspectRatioProperty); }
+            set { SetValue(AspectRatioProperty, value); }
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="AspectRatio"/> property.
+        /// </summary>
+        public static readonly DependencyProperty AspectRatioProperty =
+            DependencyProperty.Register(nameof(AspectRatio), typeof(AspectRatio), typeof(ConstrainedPresenter), new PropertyMetadata(null, AspectRatioPropertyChanged));
+
+        private static void AspectRatioPropertyChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            if (d is ConstrainedPresenter panel)
+            {
+                panel.InvalidateMeasure();
+            }
+        }
+
+        /// <summary>
+        /// Gets or sets the scale for the width of the panel. Should be a value between 0-1.0. Default is 1.0.
```

Comment:
And if it is not between 0  and 1? What if it is negative? I think both cases would cause issues with the layout-system, especially if the scale value is negative.

Reply:
Yeah, could probably coerce the value, would be nice if UWP XAML had a better system for that. Could at least clip to the bounds in the setter though.

Reply:
Yes, I think that would be a good idea. That's also what NumberBox does.